flink-yarn/src/main/java/org/apache/flink/yarn/YarnTaskManagerRunner.java

```diff
@@ -142,19 +134,22 @@ public static void runYarnTaskManager(String[] args, final Class<? extends YarnT
 
 			SecurityUtils.install(sc);
 
-			SecurityUtils.getInstalledContext().runSecured(new Callable<Object>() {
-				@Override
-				public Integer call() {
-					try {
-						TaskManager.selectNetworkInterfaceAndRunTaskManager(configuration, resourceId, taskManager);
-					}
-					catch (Throwable t) {
-						LOG.error("Error while starting the TaskManager", t);
-						System.exit(TaskManager.STARTUP_FAILURE_RETURN_CODE());
+			if (mainRunner == null) {
```

Comment:
If we need to separate out the 'secured' code block, can we do that in all scenarios instead of the conditional logic here?   For example, move the block on lines 138-147 to a new method, and pass a method reference to `runTaskManager` as the callable.

Reply:
Refactored. Since the method is only used in this class and it need access to variable in its outer scope, I just put it in the same method.